### PR TITLE
feature(registration):users endpoint

### LINKED ISSUE #52: Add users endpoint
- [ ] Admin users should be able to access the admin panel for content management
- [ ] The endpoint should be properly secured to avoid unauthorized access

### PR DESCRIPTION
**Pull request by ywalakamar**
**What does this PR do?**
- creates `api/v1/users` endpoint with the following resources:-
    - GET `api/v1/users` fetches all registered users
    - GET `api/v1/users/id` fetches a registered user with the specific id
    - POST `api/v1/users/register` creates a user
    - PATCH `api/v1/users/id` updates user details of the user with the specific id
    - DELETE `api/v1/users/id` deletes user details of the user with the specific id
- TDD tests for the `api/v1/users` endpoint
- closes #52 

**Steps to reproduce**
- Fork the repository [here](https://github.com/DevCEldoret/devceldoret-backend)
- Clone from your local repository
- Open terminal inside the cloned project root directory and:-
    - run `yarn` to install dependencies
    - run `yarn build`
    - run `yarn start` to start server
    - run `yarn test` to test

**Related Stories?**
- #52

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "server.js",
   "scripts": {
     "start": "nodemon --exec \"ts-node\" ./src/server.ts",
-    "test": "export NODE_ENV=test && nyc mocha -r ts-node/register --exit ./src/test/*.test.ts --config .mocharc.json",
+    "test": "export NODE_ENV=test && nyc mocha -r ts-node/register --exit ./src/test/*Test.ts --config .mocharc.json",
```
Comment (fatmali) [inline package.json]: `.test` or `.spec` are the standard way of naming tests. Why did you rename tests? Is there an issue you were facing?

File: src/migration/1578909700398-Initialize.ts [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
         await queryRunner.query(`CREATE TABLE "perks" ("id" SERIAL NOT NULL, "type" character varying NOT NULL, CONSTRAINT "UQ_451e535e90c45489610ae107e26" UNIQUE ("type"), CONSTRAINT "PK_7858d2e37891967413d2010a278" PRIMARY KEY ("id"))`, undefined);
-        await queryRunner.query(`CREATE TABLE "speakers" ("id" SERIAL NOT NULL, "first_name" character varying NOT NULL, "last_name" character varying NOT NULL, "gender" character varying(1) NOT NULL, "bio" character varying, "job_title" character varying NOT NULL, "image" bytea NOT NULL, CONSTRAINT "PK_b3818c94af77a0cf73403ecef14" PRIMARY KEY ("id"))`, undefined);
+        await queryRunner.query(`CREATE TABLE "speakers" ("id" SERIAL NOT NULL, "first_name" character varying NOT NULL, "last_name" character varying NOT NULL, "gender" character varying(1) NOT NULL, "bio" character varying, "job_title" character varying, "image" bytea, CONSTRAINT "PK_b3818c94af77a0cf73403ecef14" PRIMARY KEY ("id"))`, undefined);
```
Comment (fatmali) [inline src/migration/1578909700398-Initialize.ts]: it's possible we don't know or the speaker doesn't have a job, better to make `job_title` nullable

File: src/models/userModel.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  public id: string;
+
+  @Column()
+  public firstName: string;
+
+  @Column()
+  public lastName: string;
+
+  @Column()
+  public country: string;
```
Comment (fatmali) [inline src/models/userModel.ts]: `country`,`city`, not required, we need `Role`

File: package.json (hunk 1 above)
Comment (ywalakamar) [reply package.json]: On it. 

> `.test` or `.spec` are the standard way of naming tests. Why did you rename tests? Is there an issue you were facing?
Just cosmetics. Reversing it to my previous naming hehe

File: src/migration/1578909700398-Initialize.ts (hunk 2 above)
Comment (ywalakamar) [reply src/migration/1578909700398-Initialize.ts]: cool

File: src/models/userModel.ts (hunk 3 above)
Comment (ywalakamar) [reply src/models/userModel.ts]: on it!

Comment (ywalakamar) [reply src/models/userModel.ts]: On that note, how many roles should we have? I propose two, an `is_admin` column whose value is a boolean

**Comment by fatmali** [comment]
Closing this due to the wrong history of commits in this PR, I cherry-picked the commits which include the work done ontop of `development` and made a PR here #63
